// ds-alurabooks/dist/ds-alurabooks.cjs.development.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var styled = require('styled-components');
var styled__default = _interopDefault(styled);

function _taggedTemplateLiteralLoose(strings, raw) {
  if (!raw) {
    raw = strings.slice(0);
  }

  strings.raw = raw;
  return strings;
}

var _templateObject, _templateObject2, _templateObject3;
var BotaoEstilizado = /*#__PURE__*/styled__default.button(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    background: ", ";\n    padding: 16px 32px;\n    border: 2px solid #EB9B00;\n    color: ", ";\n    font-size: 20px;\n    cursor: pointer;\n    ", "\n"])), function (props) {
  return props.tipo === 'primario' ? '#EB9B00' : '#FFF';
}, function (props) {
  return props.tipo === 'primario' ? '#FFF' : '#EB9B00';
}, function (props) {
  return props.tipo === 'primario' ? styled.css(_templateObject2 || (_templateObject2 = _taggedTemplateLiteralLoose(["\n            &:hover {\n                background: #B87900;\n                border: 2px solid #B87900;\n            }\n        "]))) : styled.css(_templateObject3 || (_templateObject3 = _taggedTemplateLiteralLoose(["\n        &:hover {\n            background: #FFF;\n            border: 2px solid #B87900;\n            color: #B87900;\n        }\n        "])));
});
var AbBotao = function AbBotao(_ref) {
  var texto = _ref.texto,
      onClick = _ref.onClick,
      _ref$tipo = _ref.tipo,
      tipo = _ref$tipo === void 0 ? 'primario' : _ref$tipo;
  return React__default.createElement(BotaoEstilizado, {
    onClick: onClick,
    tipo: tipo
  }, texto);
};

var _templateObject$1;
var SectionEstilizada = /*#__PURE__*/styled__default.section(_templateObject$1 || (_templateObject$1 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    width: 194px;\n    height: 88px;\n    background: ", ";\n    border: 1px solid;\n    border-color: ", ";\n    border-radius: 8px;\n    display: flex;\n    flex-direction: column;\n    justify-content: space-around;\n    text-align: center;\n    margin: 10px;\n    font-family: sans-serif;\n    cursor: pointer;\n    header {\n        color: ", ";\n        font-size: 12px;\n        font-weight: 400;\n    }\n    strong {\n        color: ", ";;\n        font-weight: 700;\n        font-size: 16px;\n    }\n    footer {\n        color: ", ";;\n        font-weight: 400;\n        font-size: 12px;\n    }\n"])), function (props) {
  return props.selecionado ? 'linear-gradient(97.54deg, #002F52 35.49%, #326589 165.37%)' : '#FFFFFF';
}, function (props) {
  return props.selecionado ? '#002F52' : '#EB9B00';
}, function (props) {
  return props.selecionado ? '#FFF' : '#EB9B00';
}, function (props) {
  return props.selecionado ? '#FFF' : '#EB9B00';
}, function (props) {
  return props.selecionado ? '#FFF' : 'rgba(0, 0, 0, 0.54)';
});
var AbGrupoOpcoes = function AbGrupoOpcoes(_ref) {
  var opcoes = _ref.opcoes,
      onChange = _ref.onChange,
      valorPadrao = _ref.valorPadrao;

  var _useState = React.useState(valorPadrao != null ? valorPadrao : null),
      selecao = _useState[0],
      setSelecao = _useState[1];

  var aoSelecionar = function aoSelecionar(opcao) {
    setSelecao(opcao);

    if (onChange) {
      onChange(opcao);
    }
  };

  return React__default.createElement(React__default.Fragment, null, opcoes.map(function (opcao) {
    return React__default.createElement(SectionEstilizada, {
      onClick: function onClick() {
        return aoSelecionar(opcao);
      },
      key: opcao.id,
      selecionado: (selecao == null ? void 0 : selecao.id) == opcao.id
    }, React__default.createElement("header", null, opcao.titulo), React__default.createElement("div", null, React__default.createElement("strong", null, opcao.corpo)), React__default.createElement("footer", null, opcao.rodape));
  }));
};

var _templateObject$2;
var CardStyled = /*#__PURE__*/styled__default.div(_templateObject$2 || (_templateObject$2 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    padding: 48px;\n    background: #FFFFFF;\n    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);\n    border-radius: 10px;\n"])));
var AbCard = function AbCard(_ref) {
  var children = _ref.children;
  return React__default.createElement(CardStyled, null, children);
};

var _templateObject$3;
var TagStyled = /*#__PURE__*/styled__default.div(_templateObject$3 || (_templateObject$3 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    padding: 24px 32px;\n    color: #FFFFFF;\n    background: #EB9B00;\n    font-weight: 700;\n    font-size: 24px;\n    display: inline-block;\n    font-family: sans-serif;\n"])));
var AbTag = function AbTag(_ref) {
  var texto = _ref.texto;
  return React__default.createElement(TagStyled, null, texto);
};

var _templateObject$4, _templateObject2$1;
var LabelEstilizada = /*#__PURE__*/styled__default.label(_templateObject$4 || (_templateObject$4 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    color: ", ";\n    display: block;\n    font-weight: 700;\n    font-size: 16px;\n    font-family: Arial, Helvetica, sans-serif;\n    margin-left: 16px;\n    margin-bottom: 8px\n"])), function (props) {
  return props.darkmode ? '#FFF' : '#002F52';
});
var InputEstilizado = /*#__PURE__*/styled__default.input(_templateObject2$1 || (_templateObject2$1 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    font-size: 16px;\n    line-height: 24px;\n    padding: 8px 24px;\n    border: 1px solid;\n    border-radius: 45px;\n    &:focus{\n        outline: none;\n    }\n    width: 100%;\n    box-sizing: border-box;\n    background: ", ";\n    border-color: ", ";\n    text-align: ", ";\n    ::placeholder,\n    ::-webkit-input-placeholder  {\n        color: ", ";\n    }\n"])), function (props) {
  return props.darkmode ? 'transparent' : '#FFF';
}, function (props) {
  return props.darkmode ? '#FFF' : '#002F52';
}, function (props) {
  return props.placeholderAlign;
}, function (props) {
  return props.darkmode ? '#FFF' : '#002F52';
});
var AbCampoTexto = function AbCampoTexto(_ref) {
  var label = _ref.label,
      value = _ref.value,
      _onChange = _ref.onChange,
      _ref$type = _ref.type,
      type = _ref$type === void 0 ? 'text' : _ref$type,
      _ref$placeholder = _ref.placeholder,
      placeholder = _ref$placeholder === void 0 ? "" : _ref$placeholder,
      _ref$placeholderAlign = _ref.placeholderAlign,
      placeholderAlign = _ref$placeholderAlign === void 0 ? 'left' : _ref$placeholderAlign,
      _ref$darkmode = _ref.darkmode,
      darkmode = _ref$darkmode === void 0 ? false : _ref$darkmode;
  return React__default.createElement("div", null, label && React__default.createElement(LabelEstilizada, {
    darkmode: darkmode
  }, label), React__default.createElement(InputEstilizado, {
    placeholder: placeholder,
    placeholderAlign: placeholderAlign,
    darkmode: darkmode,
    type: type,
    value: value,
    onChange: function onChange(event) {
      return _onChange(event.target.value);
    }
  }));
};

var _templateObject$5, _templateObject2$2, _templateObject3$1, _templateObject4, _templateObject5;
var BotaoEstilizado$1 = /*#__PURE__*/styled__default.button(_templateObject$5 || (_templateObject$5 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    background: #002F52;\n    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.25);\n    border: none;\n    cursor: pointer;\n    color: #FFF;\n    width: 28px;\n    height: 28px;\n    border-radius: 50%;\n"])));
var SpanEstilizado = /*#__PURE__*/styled__default.span(_templateObject2$2 || (_templateObject2$2 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    font-size: 18px;\n    line-height: 27px;\n    display: inline-block;\n    margin: 0 12px;\n    color: #002F52;\n    font-family: Arial, Helvetica, sans-serif;\n"])));
var FlexContainerEstilizado = /*#__PURE__*/styled__default.div(_templateObject3$1 || (_templateObject3$1 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    display: flex;\n    align-items: center;\n    justify-content: center;\n"])));
var ContainerEstilizado = /*#__PURE__*/styled__default.div(_templateObject4 || (_templateObject4 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    width: 100px;\n    text-align: center;\n    display: inline-block;\n"])));
var LabelEstilizada$1 = /*#__PURE__*/styled__default.label(_templateObject5 || (_templateObject5 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    color: #002F52;\n    display: block;\n    font-weight: 700;\n    font-size: 16px;\n    font-family: Arial, Helvetica, sans-serif;\n    margin-bottom: 8px;\n"])));
var AbInputQuantidade = function AbInputQuantidade(_ref) {
  var onChange = _ref.onChange;

  var _useState = React.useState(1),
      value = _useState[0],
      setValue = _useState[1];

  React.useEffect(function () {
    if (onChange) {
      onChange(Number(value));
    }
  }, [value]);
  return React__default.createElement(ContainerEstilizado, null, React__default.createElement(LabelEstilizada$1, null, "Quantidade"), React__default.createElement(FlexContainerEstilizado, null, React__default.createElement(BotaoEstilizado$1, {
    onClick: function onClick() {
      return setValue(function (prev) {
        return prev - 1;
      });
    }
  }, "-"), React__default.createElement(SpanEstilizado, null, value), React__default.createElement(BotaoEstilizado$1, {
    onClick: function onClick() {
      return setValue(function (prev) {
        return prev + 1;
      });
    }
  }, "+")));
};

var _templateObject$6, _templateObject2$3, _templateObject3$2, _templateObject4$1, _templateObject5$1;
var JanelaModal = /*#__PURE__*/styled__default.div(_templateObject$6 || (_templateObject$6 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    position: fixed;\n    padding: 64px;\n    background: #FFFFFF;\n    box-shadow: 4px 4px 24px -4px rgba(0, 0, 0, 0.15);\n    border-radius: 24px;\n    top: 50%;\n    left: 50%;\n    transform: translate(-50%, -50%);\n"])));
var FundoModal = /*#__PURE__*/styled__default.div(_templateObject2$3 || (_templateObject2$3 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    position: fixed;\n    left: 0px;\n    right: 0px;\n    top: 0px;\n    bottom: 0px;\n    background: rgba(101, 101, 101, 0.85);\n"])));
var TituloModalWrapper = /*#__PURE__*/styled__default.div(_templateObject3$2 || (_templateObject3$2 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    display: flex;\n    justify-content: space-between;\n    font-family: sans-serif;\n    align-items: center;\n"])));
var TituloModal = /*#__PURE__*/styled__default.h2(_templateObject4$1 || (_templateObject4$1 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    font-weight: 700;\n    font-size: 32px;\n    line-height: 48px;\n    color: #EB9B00;\n    margin: 0;\n"])));
var BotaoFecharModal = /*#__PURE__*/styled__default.button(_templateObject5$1 || (_templateObject5$1 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    background: transparent;\n    border: none;\n    cursor: pointer;\n    color: #002F52;\n"])));
var AbModal = function AbModal(_ref) {
  var children = _ref.children,
      aberta = _ref.aberta,
      aoFechar = _ref.aoFechar,
      titulo = _ref.titulo;

  if (!aberta) {
    return React__default.createElement(React__default.Fragment, null);
  }

  return React__default.createElement(React__default.Fragment, null, React__default.createElement(FundoModal, {
    onClick: aoFechar
  }), React__default.createElement(JanelaModal, null, React__default.createElement(TituloModalWrapper, null, React__default.createElement(TituloModal, null, titulo), React__default.createElement(BotaoFecharModal, {
    onClick: aoFechar
  }, "X")), children));
};

exports.AbBotao = AbBotao;
exports.AbCampoTexto = AbCampoTexto;
exports.AbCard = AbCard;
exports.AbGrupoOpcoes = AbGrupoOpcoes;
exports.AbInputQuantidade = AbInputQuantidade;
exports.AbModal = AbModal;
exports.AbTag = AbTag;
